### ISSUE TITLE
Support for .NET Android / .NET MAUI apps

### ISSUE THREAD

**Issue by mattleibow** [DataDog/dd-sdk-android#1811]
**Are you requesting automatic instrumentation for a framework or library? Please describe.**
I am wondering about the plans and/or interest in supporting .NET for Android and/or .NET MAUI for Android.

**Is your feature request related to a problem? Please describe.**
I would like to use DataDog RUM with my .NET mobile app, but I am unable to find any bindings for the DataDog SDK for Android.

**Describe the solution you'd like**
An official or community .NET binding for the Android SDK so I ca use it in my apps.

**Describe alternatives you've considered**
I am not able to maintain a binding myself at this time.

**Additional context**

- https://learn.microsoft.com/en-us/dotnet/maui/migration/android-binding-projects
- https://learn.microsoft.com/en-us/xamarin/android/platform/binding-java-library

**Comment by MaelNamNam** [comment]
Hi Matthew and thanks for reaching out! Indeed, we are not supporting .NET frameworks at the moment, and unfortunately do not have plans to bring official support in the near future. However, it's something that has been on our radar for some time already, and I was wondering if you could go through your support channels to escalate this feature request? It will help us better track demand to re-asses our priorities, and retrieve more details about your needs (number of apps, volume of sessions, particular use cases...) in a more confidential way :)
Cheers!

**Comment by brunck** [comment]
There are .NET for Android bindings currently available via the community.

On [NuGet](nuget.org) the package IDs are `Bcr.Datadog.Android.Sdk.*`

**Comment by ognjensuljagic** [comment]
> I would like to point out that DataDog is referenced on https://learn.microsoft.com/en-us/appcenter/retirement as one the recommended "Azure Native ISV services which provide rich capabilities for mobile analytics and diagnostics." This implies that you will support .NET Android, iOS and UWP.
> 
> Are you going to provide that by the retirement deadline of March 2025?

Would appreciate an answer to this as well...

**Comment by oddfrer** [comment]
I've tested the bindings for Android from @brunck, [Bcr.Datadog.Android.Sdk.Logs](https://www.nuget.org/packages/Bcr.Datadog.Android.Sdk.Logs) and [Bcr.Datadog.Android.Sdk.Rum](https://www.nuget.org/packages/Bcr.Datadog.Android.Sdk.Rum). Everything is working well so far, but I haven’t figured out how to access the GlobalRumMonitor.

My setup:
```
static partial void InitializePlatform(string environment)
{
    // DDConfig
    var configuration = new DDConfiguration.Builder(ClientToken, environment);
    configuration.UseSite(DatadogSite.EU1!);
    Datadog.Android.Datadog.Initialize(Platform.AppContext, configuration.Build(), TrackingConsent.Granted!);
    
    // DDLogs
    var logsConfiguration = new LogsConfiguration.Builder().Build();
    Logs.Enable(logsConfiguration);

    // DDRum
    var rumConfiguration = new RumConfiguration.Builder(AppId).TrackLongTasks()
                                                              .TrackLongTasks()
                                                              .TrackFrustrations(true)
                                                              .TrackBackgroundEvents(true)
                                                              .TrackNonFatalAnrs(true)
                                                              .Build();

    Datadog.Android.Rum.Rum.Enable(rumConfiguration);
    // _rumMonitor = Datadog.Android.Rum.GlobalRumMonitor // How to access the GlobalRumMonitor?
    
    // DDLogger
    _logger = new Logger.Builder()
              .SetNetworkInfoEnabled(true)
              .SetLogcatLogsEnabled(true)
              .SetRemoteSampleRate(100f)
              .SetBundleWithTraceEnabled(true)
              .SetName("android-logger")
              .SetBundleWithRumEnabled(true)
              .Build();
}
```

Would appreciate any guidance on how to properly retrieve the GlobalRumMonitor on Android.

I've tested the [Bcr.Datadog.iOS.Sdk](https://www.nuget.org/packages/Bcr.Datadog.iOS.Sdk) and it works great! 👍

**Comment by brunck** [comment]
@oddfrer I'll take a look.

**Comment by brunck** [comment]
@oddfrer With the latest version of [Bcr.Datadog.Android.Sdk.Rum](https://www.nuget.org/packages/Bcr.Datadog.Android.Sdk.Rum/2.18.2) you can access the `GlobalRumMonitor` via `Datadog.Android.Rum.GlobalRumMonitor.Instance` or `Datadog.Android.Rum.GlobalRumMonitor.Get()`.

Note that going forward, the iOS SDK is now broken out into multiple NuGet packages so you'll want to use the [Bcr.Datadog.iOS.ObjC](https://www.nuget.org/packages/Bcr.Datadog.iOS.ObjC) and whatever other packages you need, such as `Bcr.Datadog.iOS.RUM`. [Bcr.Datadog.iOS.Sdk](https://www.nuget.org/packages/Bcr.Datadog.iOS.Sdk) won't be getting further updates.

It's pretty easy to migrate; you'll just have to add references to the packages you use and update some namespace references. The code is still the same.

**Comment by oddfrer** [comment]
Thanks, @brunck! 🎉 I’ve updated to the latest version, and accessing `GlobalRumMonitor` via `GlobalRumMonitor.Get()` works great!

Appreciate the heads-up on the iOS SDK changes as well.

Thanks again for your work on these bindings! 🚀

**Comment by JacobAtchley** [comment]
@brunck thank you for your work on the bindings! it is very much appreciated. 

i noticed on your nuget packages the link to the source repository is broken
is that because the repo is private? 

if so do you have intentions on making the repo public so other's can collaborate? 

we would like to use your package but would like to see the repo first :D

**Comment by brunck** [comment]
@JacobAtchley Yes, although it's not ready to be public quite yet. It won't be until at least next month.

**Comment by empty-set** [comment]
> [@brunck](https://github.com/brunck) thank you for your work on the bindings! it is very much appreciated.
> 
> i noticed on your nuget packages the link to the source repository is broken is that because the repo is private?
> 
> if so do you have intentions on making the repo public so other's can collaborate?
> 
> we would like to use your package but would like to see the repo first :D

We are also in a similar position. I'm looking for official support but willing to use a community solution if the binding code is available to the public. @brunck do you still have plans to open this up?

**Comment by brunck** [comment]
> Hello, 
> 
> I was wondering if you know where the public repository for the [Bcr.Datadog.Android.Sdk.Rum](https://www.nuget.org/packages/Bcr.Datadog.Android.Sdk.Rum) NuGet package is located.
> I couldn’t find any link to a GitHub repository.
> 
> This raises some concerns regarding the reliability and security of the package.
> At this point, I can’t consider using this alternative in an enterprise context without more transparency.
> 
> Thank you for any clarification you can provide.

The link should be working now.

**Comment by oddfrer** [comment]
Any update on this, @ambushwork? It looks like @brunck fixed the 16KB issue in his Bcr.Datadog.Android.Sdk.Ndk package 🙏

**Comment by kikoveiga** [comment]
Hello @mattleibow @oddfrer @craigwi @ognjensuljagic @empty-set @JacobAtchley, we're happy to announce we have finally released an official solution to support .Net and MAUI today: https://github.com/DataDog/dd-sdk-maui.
Please try it out! You can find the nuget package [here](https://www.nuget.org/packages/Datadog.Maui).
